### PR TITLE
[docs] Add glossary

### PR DESCRIPTION
**Pull request by benraskin92**
**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: docs/glossary/index.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Glossary
+
+- **Bootstrapping**: Process by which an M3DB node is brought up. Bootstrapping consists of determining
+the integrity of data that the node has, replay writes from the commit log, and/or stream missing data
+from its peers.
+
+- **Cardinality**: The number of unique metrics within the M3DB index. Cardinality increases with the
+number of unique tag/value combinations that are being emitted. 
+
+- **Datapoint**: A single timestamp/value. Timeseries are composed of multiple datapoints and a series
```
Comment (arnikola) [inline docs/glossary/index.md]: Should add basic details for `timeseries` since it's mentioned here

File: docs/glossary/index.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+of tag/value pairs. 
+
+- **Labels**: Pairs of descriptive words that give meaning to a metric. `Tags` and `Labels` are interchangeable terms.
```
Comment (arnikola) [inline docs/glossary/index.md]: Something about labels matching series dimensions?

File: docs/glossary/index.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+series database, a highly-dynamic and performant aggregation service, a query engine, and other
+supporting infrastructure.
+
+- **M3Coordinator**: A service within M3 that coordinates reads and writes between upstream systems,
+such as Prometheus, and downstream systems, such as M3DB.
+
+- **M3DB**: Distributed time series database influenced by [Gorilla](http://www.vldb.org/pvldb/vol8/p1816-teller.pdf)
+and [Cassandra](http://cassandra.apache.org/) released as open source by Uber Technologies.
+
+- **M3Query**: A distributed query engine for M3DB. Unlike M3Coordinator, M3Query only provides supports for reads.
```
Comment (arnikola) [inline docs/glossary/index.md]: m3q and m3c only differ by configs right?

File: docs/glossary/index.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+and a set of configuration options, such as data retention and block size.
+
+- **Placement**: Map of the M3DB cluster's shard replicas to nodes. Each M3DB cluster has only one placement.
+
+- **Shard**: Effectively the same as a "virtual shard" in Cassandra in that it provides an arbitrary
+distribution of time series data via a simple hash of the series ID.
+
+- **Tags**: Pairs of descriptive words that give meaning to a metric. `Tags` and `Labels` are interchangeable terms.
+
+- **Topology**: See *Placement*
```
Comment (arnikola) [inline docs/glossary/index.md]: Might be better to have
`Map of the M3DB cluster's shard replicas to nodes. Each M3DB cluster has only one placement. Topology and Placement are interchangeable terms`

File: docs/glossary/index.md (hunk 3 above)
Comment (benraskin92) [reply docs/glossary/index.md]: I don't think we need to specify that here.

Comment (anandsinghkunwar) [reply docs/glossary/index.md]: M3Query in my use can also be used for changing configuration of placements/adding nodes, namespaces.

**Pull request merged**